liblwgeom/cunit/cu_clip_by_rect.c

```diff
@@ -29,32 +29,28 @@ static void test_lwgeom_clip_by_rect(void)
 	in = lwgeom_from_wkt(wkt, LW_PARSER_CHECK_NONE);
 	out = lwgeom_clip_by_rect(in, 5, 0, 10, 10);
 	tmp = lwgeom_to_ewkt(out);
-	/* printf("%s\n", tmp); */
-	CU_ASSERT_STRING_EQUAL("LINESTRING(5 5,10 0)", tmp)
+	ASSERT_STRING_EQUAL("LINESTRING(5 5,10 0)", tmp)
 	lwfree(tmp); lwgeom_free(out); lwgeom_free(in);
 
 	wkt = "LINESTRING EMPTY";
 	in = lwgeom_from_wkt(wkt, LW_PARSER_CHECK_NONE);
 	out = lwgeom_clip_by_rect(in, 5, 0, 10, 10);
 	tmp = lwgeom_to_ewkt(out);
-	/* printf("%s\n", tmp); */
-	CU_ASSERT_STRING_EQUAL(wkt, tmp)
+	ASSERT_STRING_EQUAL(wkt, tmp)
 	lwfree(tmp); lwgeom_free(out); lwgeom_free(in);
 
 	wkt = "MULTIPOINT EMPTY";
 	in = lwgeom_from_wkt(wkt, LW_PARSER_CHECK_NONE);
 	out = lwgeom_clip_by_rect(in, 5, 0, 10, 10);
 	tmp = lwgeom_to_ewkt(out);
-	/* printf("%s\n", tmp); */
-	CU_ASSERT_STRING_EQUAL(wkt, tmp)
+	ASSERT_STRING_EQUAL(wkt, tmp)
 	lwfree(tmp); lwgeom_free(out); lwgeom_free(in);
 
 	wkt = "MULTIPOINT(0 0, 6 6, 7 5)";
 	in = lwgeom_from_wkt(wkt, LW_PARSER_CHECK_NONE);
 	out = lwgeom_clip_by_rect(in, 5, 0, 10, 10);
 	tmp = lwgeom_to_ewkt(out);
-	/* printf("%s\n", tmp); */
-	CU_ASSERT_STRING_EQUAL("MULTIPOINT(6 6,7 5)", tmp)
+	ASSERT_STRING_EQUAL("MULTIPOINT(6 6,7 5)", tmp)
```

Comment:
Why this change?  It seems like there are a lot of these, and they are not related to the output precision issue?

Reply:
It's true. Due to how I ended up dividing the PRs I squashed multiple commits into one and I lost quite a bit of information that I should recover before merging.

`CU_ASSERT_STRING_EQUAL` compares strings and returns an error, but it doesn't output the values (so you are blind as you what the difference was). OTOH, we also have `ASSERT_STRING_EQUAL` which does the same but outputs the strings when an error happens. Since there were multiple broken tests that didn't output the diffs, I decided to replace all calls to the one with the output and removed any manual printf that some tests had to do that.

Reply:
Makes sense.  But does make the diff seem larger than it actually is.

Reply:
You are right. I'll try to break down the PR into different commits or even remove that change and leave only the digit changes so it's more clear what's truly changing in the PR.

Reply:
I'm going to force-push removing these changes and leaving only the minimum changes reflecting the `return always N amount of decimal digits` change.